imagebuildah/build.go

```diff
@@ -721,32 +698,25 @@ func BuildDockerfiles(store storage.Store, options BuildOptions, dockerfile ...s
 				resp.Body.Close()
 				return errors.Errorf("no contents in %q", dfile)
 			}
-			rc = resp.Body
+			parsed, err = imagebuilder.ParseDockerfile(resp.Body)
+			resp.Body.Close()
 		} else {
 			if !filepath.IsAbs(dfile) {
 				logrus.Debugf("resolving local Dockerfile %q", dfile)
 				dfile = filepath.Join(options.ContextDirectory, dfile)
 			}
 			logrus.Debugf("reading local Dockerfile %q", dfile)
-			contents, err := os.Open(dfile)
-			if err != nil {
-				return errors.Wrapf(err, "error reading %q", dfile)
-			}
-			dinfo, err := contents.Stat()
-			if err != nil {
-				contents.Close()
-				return errors.Wrapf(err, "error reading info about %q", dfile)
-			}
-			if dinfo.Size() == 0 {
-				contents.Close()
-				return errors.Wrapf(err, "no contents in %q", dfile)
-			}
-			rc = contents
+			parsed, err = imagebuilder.ParseFile(dfile)
+		}
+		if err != nil {
+			return errors.Wrapf(err, "uname to parse %s", dfile)
```

Comment:
That error message looks odd.  Should it be `"unable to parse %s"`?

Reply:
fixed